include/deal.II/physics/notation.h

```diff
@@ -0,0 +1,1550 @@
+      Vector<Number>
+      unroll_to_vector (const Number &s)
+      {
+        Vector<Number> out (1);
+        const unsigned int n_rows = out.size();
+        for (unsigned int r=0; r<n_rows; ++r)
+          out(r) = s;
+        return out;
+      }
+
+
+      template<int dim, typename Number>
+      Vector<Number>
+      unroll_to_vector (const Tensor<0,dim,Number> &s)
+      {
+        return unroll_to_vector(s.operator const Number &());
+      }
+
+
+      template<int dim, typename Number>
+      Vector<Number>
+      unroll_to_vector (const Tensor<1,dim,Number> &v)
+      {
+        Vector<Number> out (v.n_independent_components);
+        const unsigned int n_rows = out.size();
+        for (unsigned int r=0; r<n_rows; ++r)
+          {
+            const std::pair<unsigned int,unsigned int> indices = internal::indices_from_component<dim>(r,false);
+            Assert(indices.first < dim, ExcInternalError());
+            const unsigned int &i = indices.first;
+            out(r) = v[i];
+          }
+        return out;
+      }
+
+
+      template<int dim, typename Number>
+      Vector<Number>
+      unroll_to_vector (const Tensor<2,dim,Number> &t)
+      {
+        Vector<Number> out (t.n_independent_components);
+        const unsigned int n_rows = out.size();
+        for (unsigned int r=0; r<n_rows; ++r)
+          {
+            const std::pair<unsigned int,unsigned int> indices = internal::indices_from_component<dim>(r,false);
+            Assert(indices.first < dim, ExcInternalError());
+            Assert(indices.second < dim, ExcInternalError());
+            const unsigned int &i = indices.first;
+            const unsigned int &j = indices.second;
+            out(r) = t[i][j];
+          }
+        return out;
+      }
+
+
+      template<int dim, typename Number>
+      Vector<Number>
+      unroll_to_vector (const SymmetricTensor<2,dim,Number> &st)
+      {
+        Vector<Number> out (st.n_independent_components);
+        const unsigned int n_rows = out.size();
+        for (unsigned int r=0; r<n_rows; ++r)
+          {
+            const std::pair<unsigned int,unsigned int> indices = internal::indices_from_component<dim>(r,true);
+            Assert(indices.first < dim, ExcInternalError());
+            Assert(indices.second < dim, ExcInternalError());
+            Assert(indices.second >= indices.first, ExcInternalError());
+            const unsigned int &i = indices.first;
+            const unsigned int &j = indices.second;
+
+            const double factor = internal::vector_component_factor<dim>(r,true);
+
+            out(r) = factor*st[i][j];
+          }
+        return out;
+      }
+
+
+      template<int,typename Number>
+      FullMatrix<Number>
+      unroll_to_matrix (const Number &s)
+      {
+        FullMatrix<Number> out (1,1);
+        const unsigned int n_rows = out.m();
+        const unsigned int n_cols = out.n();
+        for (unsigned int r=0; r<n_rows; ++r)
```

Comment:
this all seems unnecessarily complicated -- I know it's just copy-paste, but you can replace lines 874-884 by
```
  out(1,1) = s;
```
:-)

Reply:
Are you testing whether I'm awake? `out(0,0) = s;` ;-)

Reply:
Haha! :-)